### ISSUE TITLE
TypeError: can not serialize 'DocTransformerOutput' object

### ISSUE THREAD

**Issue by sliedes** [explosion/spaCy#13238]
This seems to be exactly #6672, but since that's locked, I cannot comment on it.

## How to reproduce the behaviour

The example from #6672:

```python
import spacy
sentence = "I love you."
nlp = spacy.load('en_core_web_trf')
doc = nlp(sentence)
doc.to_bytes()
```

raises `TypeError: can not serialize 'DocTransformerOutput' object`.

## Info about spaCy

- **spaCy version:** 3.7.2
- **Platform:** Linux-6.2.0-1018-lowlatency-x86_64-with-glibc2.37
- **Python version:** 3.11.4
- **Pipelines:** en_core_web_trf (3.7.3), de_core_news_lg (3.7.0), de_core_news_md (3.7.0), de_core_news_sm (3.7.0), de_dep_news_trf (3.7.2)

**Comment by svlandeg** [comment]
Thanks for the report!

It's probably a different issue than #6672 (though they look similar). Under the hood, spaCy 3.7 now uses `curated-transformers` for the `trf` pipelines, assigning a `DocTransformerOutput` object to `doc._.trf_data`. This is where your error seems to originate from.

FYI - for spaCy v3.0-v3.6, the `trf` pipelines were storing a `TransformerData` object in `doc._.trf_data`, and I think this one serialized just fine. To double check, you can always try running the same code (with a compatible trf model) with spaCy 3.6.

Anyway, we need to look into this. It's related to https://github.com/explosion/spaCy/issues/13233, as well.

**Comment by svlandeg** [comment]
Update: should be fixed by https://github.com/explosion/spacy-curated-transformers/pull/29 🤞

**Comment by danieldk** [comment]
We have released `spacy-curated-transformers` 0.2.2, a `pip install --upgrade spacy-curated-transformers` should fix this issue.

Thanks again for reporting this issue and feel free to re-open it if the updated package doesn't solve the issue for you.
